meteor/client/lib/ShowStyleVariantImportButton.tsx

```
@@ -0,0 +1,168 @@
+				}
+			})
+		}
+
+		private showStyleVariantAlreadyExists(importedShowStyleVariant: ShowStyleVariant): boolean {
+			const exists = this.props.showStyleVariants.find(
+				(variant: ShowStyleVariant) => variant.name === importedShowStyleVariant.name
+			)
+			return !!exists
+		}
+
+		private provideImportOptions(showStyleVariant: ShowStyleVariant) {
+			doModalDialog({
+				title: t('Do you want to replace this variant?'),
+				yes: t('Replace'),
+				no: t('Keep both variants'),
+				onAccept: () => {
+					this.replaceShowStyleVariant(showStyleVariant)
+				},
+				onSecondary: () => {
+					const resemblingNameCount = this.getDuplicatedShowStyleVariantAmount(showStyleVariant)
+					showStyleVariant.name += ' (' + resemblingNameCount + ')'
+					this.importShowStyleVariant(showStyleVariant)
+				},
+				message: (
+					<React.Fragment>
+						<p>
+							{t('Do you want to replace "{{showStyleVariantName}}"?', {
+								showStyleVariantName: showStyleVariant.name,
+							})}
+						</p>
+					</React.Fragment>
+				),
+			})
+		}
+
+		private importShowStyleVariant(showStyleVariant: ShowStyleVariant) {
+			MeteorCall.showstyles.importShowStyleVariantAsNew(showStyleVariant).catch(() => {
+				NotificationCenter.push(
+					new Notification(
+						undefined,
+						NoticeLevel.WARNING,
+						t('Failed to import Variant {{name}}.', {
+							name: showStyleVariant.name,
+						}),
+						'VariantSettings'
+					)
+				)
+			})
+		}
+
+		private replaceShowStyleVariant(newShowStyleVariant: ShowStyleVariant) {
+			const existingShowStyleVariant = this.props.showStyleVariants.find((variant: ShowStyleVariant) => {
+				return variant.name === newShowStyleVariant.name
+			})
+			if (existingShowStyleVariant) {
+				MeteorCall.showstyles
+					.removeShowStyleVariant(existingShowStyleVariant._id)
+					.then(() => {
+						MeteorCall.showstyles.importShowStyleVariantAsNew(newShowStyleVariant).catch(logger.warn)
+					})
+					.catch(logger.warn)
```

Should we just silently catch or should there be a notification shown?

---

Good catch, since we do it in the other scenario by using the `importShowStyleVariant` method, this was an oversight. I will instead use that method which includes the notification.